subsystem/core/src/main/java/org/wildfly/extension/camel/service/CamelEndpointDeploymentSchedulerService.java

```diff
@@ -0,0 +1,181 @@
+        final CamelEndpointDeploymentSchedulerService service = new CamelEndpointDeploymentSchedulerService(deploymentName);
+        return serviceTarget.addService(deploymentSchedulerServiceName(deploymentUnitServiceName), service) //
+                .install();
+    }
+
+    public static ServiceName deploymentSchedulerServiceName(ClassLoader deploymentClassLoader) {
+        if (deploymentClassLoader == null) {
+            deploymentClassLoader = SecurityActions.getContextClassLoader();
+        }
+        if (deploymentClassLoader instanceof ModuleClassLoader) {
+            ModuleClassLoader moduleClassLoader = (ModuleClassLoader) deploymentClassLoader;
+            final String DEPLOYMEND_CL_NAME_PREFIX = "deployment.";
+            final String clName = moduleClassLoader.getName();
+            if (clName.startsWith(DEPLOYMEND_CL_NAME_PREFIX)) {
+                final String deploymentName = clName.substring(DEPLOYMEND_CL_NAME_PREFIX.length());
+                final int earInfixPos = deploymentName.indexOf(EAR_INFIX);
+                if (earInfixPos >= 0) {
+                    final String earName = deploymentName.substring(0, earInfixPos + EAR_INFIX.length() - 1);
+                    final String warName = deploymentName.substring(earInfixPos + EAR_INFIX.length());
+                    return ServiceName.of("jboss", "deployment", "subunit", earName, warName, SERVICE_NAME);
+                } else {
+                    return ServiceName.of("jboss", "deployment", "unit", deploymentName, SERVICE_NAME);
+                }
+            } else {
+                throw new IllegalStateException(String.format("Expected a %s name starting with '%s'; found %s", ModuleClassLoader.class.getName(), DEPLOYMEND_CL_NAME_PREFIX, clName));
+            }
+        } else {
+            throw new IllegalStateException(String.format("Expected a %s; found %s", ModuleClassLoader.class.getName(), deploymentClassLoader));
+        }
+    }
+    public static ServiceName deploymentSchedulerServiceName(ServiceName deploymentUnitServiceName) {
+        return deploymentUnitServiceName.append(SERVICE_NAME);
+    }
+    private volatile CamelEndpointDeployerService deployerService;
+
+    private final String deploymentName;
+
+    private final Map<URI, EndpointHttpHandler> scheduledHandlers = new HashMap<>();
+
+    CamelEndpointDeploymentSchedulerService(String deploymentName) {
+        super();
+        this.deploymentName = deploymentName;
+    }
+
+    @Override
+    public CamelEndpointDeploymentSchedulerService getValue() throws IllegalStateException {
+        return this;
+    }
+
+    /**
+     * Either schedules the given HTTP endpoint for deployment once {@link #deployerService} becomes available or
+     * deploys it instantly if the {@link #deployerService} is available already.
+     *
+     * @param uri
+     *                                determines the path and protocol under which the HTTP endpoint should be exposed
```

Comment:
Noticed in a few places, the Javadoc indentation / line continuation seems a bit weird.

Reply:
Let me check if Eclipse formatter can be persuaded not to do that.

Reply:
Yeah, found some settings. Is this OK?

```
     * @param uri determines the path and protocol under which the HTTP endpoint should be exposed
     * @param endpointHttpHandler an {@link EndpointHttpHandler} to use for handling HTTP requests sent to the given
     *        {@link URI}'s path
```

Reply:
Still find it weird that it leaves random whitespace on the new line. But hey, it's an improvement..

Reply:
> it leaves random whitespace on the new line

It is aligned with the first word after `@param`. Apparently this is the first multiline param comment in the whole WFC codebase so I dare to declare this as the new official convention :)